### ISSUE TITLE
Issue with initial run

### ISSUE THREAD

**Issue by gswalden** [tngan/express-saml2#29]
I'm having an issue getting v1.2.6 to run on my system, and was hoping the error was familiar enough to allow an easy suggestion. Here are the steps I took:
1. Cloned the repo
2. Ran `npm install` in the main directory, then ran that command again in `examples/idp/`, `examples/sp1/`, and `examples/sp2/`
3. Ran `node app.js` for each of the 3 folders above
4. Successfully logged into http://localhost:3001, then clicked `Sign in application 1 (:4002)`
5. Received error: 
```
Error: Undefined assertion or invalid syntax
    at Object.decryptAssertion (/Users/givemesnacks/repo/express-saml2/lib/SamlLib.js:509:52)
    at Entity.abstractBindingParser (/Users/givemesnacks/repo/express-saml2/lib/Entity.js:202:21)
    at Entity.parseLoginResponse (/Users/givemesnacks/repo/express-saml2/lib/ServiceProvider.js:80:21)
    at /Users/givemesnacks/repo/express-saml2/examples/sp1/routes/sso.js:100:9
    at Layer.handle [as handle_request] (/Users/givemesnacks/repo/express-saml2/examples/sp1/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/givemesnacks/repo/express-saml2/examples/sp1/node_modules/express/lib/router/route.js:131:13)
    at Route.dispatch (/Users/givemesnacks/repo/express-saml2/examples/sp1/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/Users/givemesnacks/repo/express-saml2/examples/sp1/node_modules/express/lib/router/layer.js:95:5)
    at /Users/givemesnacks/repo/express-saml2/examples/sp1/node_modules/express/lib/router/index.js:277:22
    at param (/Users/givemesnacks/repo/express-saml2/examples/sp1/node_modules/express/lib/router/index.js:349:14)
```

I am running macOS 10.12.1 and Node v7.2.1 (I also tried Node v5.12.0 but received the same result). Thanks @tngan for any help you can provide!

**Comment by tngan** [comment]
@gswalden Thanks for your report. Have you run `npm install` in each of example folder ? I have tested with Node v7.2.1 and v6.9.2, and it works properly.

I have updated version of dependencies in example, you can now test with v1.2.7.

**Comment by gswalden** [comment]
Which OS are you using? I experienced the same error on OS X El Capitan, running v1.2.7. Looking at the code, the error is thrown when the XML signature is being examined, which makes me think something related to the encryption is breaking (which could be OS-dependent).

**Comment by tngan** [comment]
@gswalden i am running on macOS 10.12

**Comment by gswalden** [comment]
I was able to get it working if I disabled encryption.

I believe I found the issue with encryption, not sure what the underlying cause is (or if the issue is unique to me). This [`.replace()` fails](https://github.com/tngan/express-saml2/blob/master/lib/SamlLib.js#L476), as the `saml:Assertion` block in `entireXML` is different than what is in the variable `assertion`. [This diff](https://www.diffchecker.com/oh8z95SO) shows the difference (left is variable `assertion`, right side is extracted from `entireXML`).

Thanks for the library, I'm able to continue forward for now 👍

**Comment by tngan** [comment]
@gswalden Would you like to provide the code you have modified in case you have made changes in example folder ? I am happy to help. 

Besides, is the same issue replicated when you click `Sign in application 2` after you login to idp ?

**Comment by gswalden** [comment]
Sorry for the delayed response, here we go:

Replacing ([src](https://github.com/tngan/express-saml2/blob/master/lib/SamlLib.js#L476))
```js
callback(Utility.base64Encode(entireXML.replace(assertion, '<saml:EncryptedAssertion>' + res + '</saml:EncryptedAssertion>')));
```

with
```js
callback(Utility.base64Encode(entireXML.replace(/<saml:Assertion.+saml:Assertion>/, '<saml:EncryptedAssertion>' + res + '</saml:EncryptedAssertion>')));
```

fixed the issue for me. I'm sure there is a better way to handle this (parsing HTML with regex is generally frowned upon; maybe this function [replaceTagsByValue](https://github.com/tngan/express-saml2/blob/master/lib/SamlLib.js#L269) would be preferred?) but I don't have enough experience with XML libraries to know offhand.

I'd be happy to make a pull request with this change, or to discuss a better method.

**Comment by tngan** [comment]
@gswalden `replaceTagsByValue` is used in when you want to customize the template, and I think `.replace` should be good enough to do the task. What's the result when you run `npm test` without making the above change ?

I have doubled checked your diff above, and `xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion"` is missing in the Assertation section of `entireXML`, and I discovered `xmlns:saml` is also missing in default login response template ([here](https://github.com/tngan/express-saml2/blob/master/lib/SamlLib.js#L257)), let's see if it works after you put it back. I am always happy to review and accept pull requests.
